mlflow/models/evaluation/base.py

```diff
@@ -161,7 +164,7 @@ class EvaluationDataset:
     NUM_SAMPLE_ROWS_FOR_HASH = 5
     SPARK_DATAFRAME_LIMIT = 10000
 
-    def __init__(self, data, labels, name=None, path=None):
+    def __init__(self, data, labels, name=None, path=None, feature_names=None):
```

Comment:
Not specific to this addition.

It might be a little late, but what's the reason to support so many different ways of specifying the data and labels input? Specifically, when data is an ndarray, it **should not** contain label column, but if data is a DataFrame, it **should**. Isn't this really confusing and error prune? If we just require data to be a DataFrame, and label a text string for the column name in the data DF, there's no need to specify feature_names (as DataFrame contains it already) and all the complex check logics below is not necessary.

Reply:
Oh, `feature_names` is not designed to specify which columns in dataset are "feature" columns, it is used to append feature names to ndarray input. The shap explainability plot will show feature names, for ndarray input case, if no feature name provided, we will use `f_{i}` instead, which is less readable.